src/sentry/search/snuba/backend.py

```diff
@@ -341,6 +285,34 @@ def _query(self, projects, retention_window_start, group_queryset, tags, environ
         max_time = options.get('snuba.search.max-total-chunk-time-seconds')
         time_start = time.time()
 
+        if count_hits and candidate_ids is None:
+            # If we have no candidates, get a random sample of groups matching
+            # the snuba side of the query, and see how many of those pass the
+            # post-filter in postgres. This should give us an estimate of the
+            # total number of snuba matches that will be overall matches, which
+            # we can use to get an estimate for X-Hits. Note no cursor, so we
+            # are always estimating the total hits.
+            snuba_groups, snuba_total = snuba_search(
+                start=start,
+                end=end,
+                project_ids=[p.id for p in projects],
+                environment_ids=environments and [environment.id for environment in environments],
+                tags=tags,
+                sort_field=sort_field,
+                get_sample=True,
+                **parameters
+            )
+            snuba_count = len(snuba_groups)
+            if snuba_count == 0:
+                return EMPTY_RESULT
```

Comment:
To make sure I understand this correctly: in the case where there is a potentially a very large result set from the Postgres side (e.g. using `is:unresolved`), but we're searching for something super specific on the Snuba side (e.g. a high cardinality tag value), this will return an empty result if that item on the Snuba side isn't part of the population sample? (Is there a workaround that I overlooked?)

Reply:
I think I wrote this comment before really understanding (or even reading) the sample stuff in my head and thought this was actually doing a `SAMPLE` clause. This makes a lot more sense now.